common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -60,6 +60,33 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
 
   }
 
+  object AmpSoundcloud {
+    def createElement(document: Document, trackId: String): Element = {
+      val soundcloud = document.createElement("amp-soundcloud")
+      soundcloud.attr("data-trackid", trackId)
+      soundcloud.attr("data-visual", "true")
+      soundcloud.attr("height", "300") // height is necessary if data-visual == true
+    }
+
+    def getTrackIdFromUrl(soundcloudUrl: String): String = {
+      val pattern = "soundcloud.com%2Ftracks%2F(\\d+)".r
+      pattern.findAllIn(soundcloudUrl).matchData.map { matches => matches.group(1) }.mkString
```

Comment:
Why not doing a simple match rather than a findAllIn -> map -> group(1) -> mkString?
Also should this method return a Option[String] to clearly signify there is no result rather than relying on an empty string